File: app/src/main/java/com/returnnotfound/stain/utils/FontUtils.java
package com.returnnotfound.stain.utils;


import android.content.Context;
import android.graphics.Typeface;

import java.util.Hashtable;

public abstract class FontUtils {

  private static final Hashtable<String, Typeface> CACHE = new Hashtable<>();

  public static Typeface get(Context c, String font) {
    synchronized (CACHE) {
      if (!CACHE.containsKey(font)) {
        try {
          String path = "fonts/" + font;
          Typeface t = Typeface.createFromAsset(c.getAssets(), path);
          CACHE.put(font, t);
        } catch (Exception e) {
          return null;
        }
      }
      return CACHE.get(font);
    }
  }
}

File: app/src/main/java/com/returnnotfound/stain/base/utils/FontUtils.kt
package com.returnnotfound.stain.base.utils

import android.content.Context
import android.graphics.Typeface
import java.util.*

object FontUtils {
  private val CACHE = Hashtable<String, Typeface>()

  @JvmStatic
  operator fun get(c: Context, path: String): Typeface? {
    synchronized(CACHE) {
      if (!CACHE.containsKey(path)) {
        try {
          val t = Typeface.createFromAsset(c.assets, path)
          CACHE[path] = t
        } catch (e: Exception) {
          return null
        }
      }
      return CACHE[path]
    }
  }
}